modules/admin-bar/module.php

```diff
@@ -0,0 +1,132 @@
+<?php
+namespace Elementor\Modules\AdminBar;
+
+use Elementor\Plugin;
+use Elementor\Core\Base\Document;
+use Elementor\Core\Base\Module as BaseModule;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Module extends BaseModule {
+	/**
+	 * @var Document[]
+	 */
+	private $documents = [];
+
+	/**
+	 * @return bool
+	 */
+	public static function is_active() {
+		return is_admin_bar_showing();
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_name() {
+		return 'admin-bar';
+	}
+
+	/**
+	 * Collect the documents that was rendered in the current page.
+	 *
+	 * @param Document $document
+	 * @param $is_excerpt
+	 */
+	public function add_document_to_admin_bar( Document $document, $is_excerpt ) {
+		if (
+			$is_excerpt ||
+			! $document::get_property( 'show_on_admin_bar' ) ||
+			! $document->is_editable_by_current_user()
+		) {
+			return;
+		}
+
+		$this->documents[ $document->get_main_id() ] = $document;
+	}
+
+	/**
+	 * Scripts for module.
+	 */
+	public function enqueue_scripts() {
+		if ( empty( $this->documents ) ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			'elementor-admin-bar',
+			$this->get_js_assets_url( 'admin-bar' ),
+			[ 'elementor-common' ],
+			ELEMENTOR_VERSION,
+			true
+		);
+
+		wp_localize_script(
+			'elementor-admin-bar',
+			'elementor_admin_bar',
+			$this->get_admin_bar_config()
+		);
```

Comment:
If the admin-bar module uses its own config, it's better to extend it from `Core\Base\App`

Reply:
As above it's settings for the client side.

Reply:
@Nevoss use App->print_config( 'elementor-admin-bar' );